src/ddmd/dmodule.d

```diff
@@ -528,6 +532,24 @@ extern (C++) final class Module : Package
             fprintf(global.stdmsg, "%s\t(%s)\n", ident.toChars(), m.srcfile.toChars());
         }
         m = m.parse();
+
+        // set the compileImport attribute here if applicable.
+        // need to do it early because it affects the semantic analysis, however,
+        // it can only be done after parsing because determining whether an import
+        // should be compiled requires the module name which is only known once
```

Comment:
Note to self.  Actually, the module name should be known before parsing because this only applies to "imported modules" and all imported module names can be derived from the import that caused the module to be loaded.  So you could set this attribute before parsing the module, but currently I didn't see any case where the code needed to know if a module was going to have it's object code generated, so it should be fine to wait until after the module has been parsed.